Tribler/Core/TorrentChecker/torrent_checker.py

```diff
@@ -406,11 +422,8 @@ def _update_torrent_result(self, response):
         torrent_id = result[u'torrent_id']
         retries = result[u'tracker_check_retries']
 
-        # the result logic
-        is_good_result = seeders > 0 or leechers > 0
-
         # the status logic
-        if is_good_result:
+        if seeders > 0:
```

Comment:
RE lying trackers: We are already keeping the best response across all trackers for a given torrent (torrent_checker.py:407)

The issue comes when we randomly select a torrent to fill a multiscrape request for a specific tracker triggered by a manually selected tracker (from the GUI). We will then be only checking one of the (possibly) multiple trackers it contains and still increasing the retry counter for it. One possible solution would be to mark those randomly selected torrents as such so we can avoid increasing the retry counter. The other would be that just before marking a torrent as dead, we should do a final full check on it and only then mark it at dead so it can be purged from the system.

Also I guess it's a good moment to re-add DHT support to the checker.